cmd/batch-expire.go

```diff
@@ -286,24 +287,31 @@ type BatchJobExpire struct {
 	Rules           []BatchJobExpireFilter `yaml:"rules" json:"rules"`
 }
 
-var _ yaml.Unmarshaler = &BatchJobExpire{}
+// BatchJobExpireV2 represents configuration parameters for a batch expiration
+// job typically supplied in yaml form
+type BatchJobExpireV2 struct {
+	BatchJobExpireV1
```

Comment:
@jiuker I see that you have used struct embedding in multiple places in this PR. While it saves typing common fields it has the following downsides.
- Shadowing of struct fields makes it harder to read surrounding code for someone new to this feature (maintainability)
- Shadowing affects how various libraries marshal/unmarshal these struct values; It's best to avoid surprises.

I suggest we use separate types for job v1 and v2 without any struct embedding for the sake of reducing repeated fields.